### PR TITLE
Feat/1022 co edited

### PR DESCRIPTION
**Pull request by zjimf**
請先下載 redis，並執行`redis-server`，開啟 redis

各 api 所需參數皆寫於 swagger 文件中，可於 [http://localhost:3001/api-docs](http://localhost:3001/api-docs) 開啟查看

**space**
    - GET /space/getSpaceDetails: for get user all space
    - POST /space/createSpace: for create space
    - DELETE space/deleteSpace: for delete space

**block**
    - GET /block/{block id}: get block info by id
    - GET /block/{space id}/blocks: get blocks id by space
    - POST /block/{space id}/createBlock: create block in specific space
    - POST /block/{block id}/generatePin: generate pin code in block
    - POST /block/join: join co work block
    - POST block/initializeBlockSocket: build socket to block
    - PUT /block/{block id}: save block information

block/initializeBlockSocket 此 api 較特別，對 block 開啟 socket 通道，開啟後可至 [WebSocket King](https://websocketking.com/) 開通 websocket 對 block 的連結，連結為ws://localhost:3001?workspaceId={block's Id}，即可連接，可開通多個通道至同一個 block，以測試redis pub/sub 功能

### CODE REVIEW

**Comment by HenryHung90** [comment]
另外這邊是否有辦法整理成使用 RESTful 架構的 api ? 這樣應該可以提高可讀性～

**Comment by zjimf** [comment]
Ok I'll do it, Tks!

**Comment by HenryHung90** [comment]
@zjimf 上面還有幾個 comments 再幫我看一下~

**Comment by zjimf** [comment]
@HenryHung90 checked, can be merged now

**Review by HenryHung90 — APPROVED** [review]

File: docker-compose.yaml [hunk 1]
```diff
@@ -1,27 +1,31 @@
 services:
   stamind-backend1:
-    image: douji/stamind-backend1:latest  # 使用推送到 Docker Hub 的映像檔
+    image: douji/stamind-backend1:latest # 使用推送到 Docker Hub 的映像檔
     ports:
       - "3001:3001"
     environment:
       - MONGO_URI=mongodb://mongo:27017/stamind
```
[the remark this answers is not in the document]
Comment (zjimf) [reply docker-compose.yaml:7]: 在docker容器運行的話，用 mongodb://mongo:27017/stamind沒錯哦！

File: src/config/httpsConfig.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import fs from 'fs';
+
+const certPath = '/etc/letsencrypt/live/ccj.infocom.yzu.edu.tw/fullchain.pem';
+const keyPath = '/etc/letsencrypt/live/ccj.infocom.yzu.edu.tw/privkey.pem';
```
[the remark this answers is not in the document]
Comment (zjimf) [reply src/config/httpsConfig.ts:4]: ok, 我會記起來再下一個branch塞一個commit

File: src/models/block.ts [hunk 3]
```diff
@@ -1,81 +1,27 @@
...
-// interface IBlockAITree {
-//     gptContent: object,
-// }
+import mongoose, { Schema, Document } from 'mongoose';
 
 export interface IBlock extends Document {
-    spaceId: string,
-    blockId: string,
-    blockName: string,
-    blockSetting: IBlockSetting,
```
[the remark this answers is not in the document]
Comment (zjimf) [reply src/models/block.ts:26]: 沒注意到刪掉了QQ, 我會記起來, sorry

**Pull request merged**